test/unit/inbound/process-submitted-settlement.test.js

```diff
@@ -0,0 +1,131 @@
+const mockCommit = jest.fn()
+const mockRollback = jest.fn()
+const mockTransaction = {
+  commit: mockCommit,
+  rollback: mockRollback
+}
+
+jest.mock('../../../app/data', () => {
+  return {
+    sequelize:
+       {
+         transaction: jest.fn().mockImplementation(() => {
+           return { ...mockTransaction }
+         })
+       }
+  }
+})
+
+jest.mock('../../../app/inbound/get-settlement-by-invoicenumber-and-value')
+const getSettlementByInvoiceNumberAndValue = require('../../../app/inbound/get-settlement-by-invoicenumber-and-value')
+
+jest.mock('../../../app/inbound/save-settlement')
+const saveSettlement = require('../../../app/inbound/save-settlement')
+
+const processReturnSettlement = require('../../../app/inbound/process-return-settlement')
+const settlement = JSON.parse(JSON.stringify(require('../../mock-settlement')))
+
+describe('process return settlement request', () => {
+  beforeEach(() => {
+    getSettlementByInvoiceNumberAndValue.mockResolvedValue(null)
+    saveSettlement.mockResolvedValue(null)
+  })
+
+  afterEach(() => {
+    jest.clearAllMocks()
+  })
+
+  test('should call getSettlementByInvoiceNumberAndValue when a valid settlement is given', async () => {
+    await processReturnSettlement(settlement)
+    expect(getSettlementByInvoiceNumberAndValue).toBeCalled()
+  })
+
+  test('should call getSettlementByInvoiceNumberAndValue once when a valid settlement is given', async () => {
+    await processReturnSettlement(settlement)
+    expect(getSettlementByInvoiceNumberAndValue).toBeCalledTimes(1)
+  })
+
+  test('should call getSettlementByInvoiceNumberAndValue with "settlement" and "mockTransaction" when valid settlement is given', async () => {
+    await processReturnSettlement(settlement)
+    expect(getSettlementByInvoiceNumberAndValue).toHaveBeenCalledWith(settlement.invoiceNumber, settlement.value, mockTransaction)
+  })
+
+  test('should throw when getSettlementByInvoiceNumberAndValue throws', async () => {
+    getSettlementByInvoiceNumberAndValue.mockRejectedValue(new Error('Database retrieval issue'))
+    const wrapper = async () => {
+      await processReturnSettlement(settlement)
+    }
+    expect(wrapper).rejects.toThrow()
+  })
+
+  test('should throw an error of "Database retrieval issue" when getSettlementByInvoiceNumberAndValue throws', async () => {
+    getSettlementByInvoiceNumberAndValue.mockRejectedValue(new Error('Database retrieval issue'))
+    const wrapper = async () => {
+      await processReturnSettlement(settlement)
+    }
+    expect(wrapper).rejects.toThrow(Error('Database retrieval issue'))
+  })
+
+  test('should call mockTransaction.rollback when a duplicate settlement is found', async () => {
+    const duplicateSettlement = settlement
+    getSettlementByInvoiceNumberAndValue.mockResolvedValue(duplicateSettlement)
```

Comment:
`getSettlementByInvoiceNumberAndValue.mockResolvedValue(settlement)`

then you can get rid of line 70

Reply:
same for next few tests below